File: Python_Programs/file_copy.py
import shutil
import sys
from pathlib import Path

def copy_file(source: Path, dest: Path) -> None:
    """Copy a file from source to destination."""
    try:
        shutil.copy2(source, dest)
        print(f"Successfully copied {source} -> {dest}")
    except Exception as e:
        print(f"Error: {e}", file=sys.stderr)
        sys.exit(1)

def main():
    if len(sys.argv) != 3:
        print(f"Usage: {sys.argv[0]} source_file dest_file", file=sys.stderr)
        sys.exit(1)
    
    source = Path(sys.argv[1])
    dest = Path(sys.argv[2])
    copy_file(source, dest)

if __name__ == "__main__":
    main()

File: Java_Programs/FileCopy.java
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.StandardCopyOption;

public class FileCopy {
    public static void main(String[] args) throws Exception {
        if (args.length != 2) {
            System.err.println("Usage: java FileCopy sourceFile destFile");
            System.exit(1);
        }
        Path src = Path.of(args[0]);
        Path dst = Path.of(args[1]);
        Files.copy(src, dst, StandardCopyOption.REPLACE_EXISTING);
        System.out.println("Copied " + src + " -> " + dst);
    }
}
